pmd-visualforce/src/main/java/net/sourceforge/pmd/lang/vf/rule/security/VfCsrfRule.java

```diff
@@ -0,0 +1,67 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.lang.vf.rule.security;
+
+import java.util.List;
+
+import net.sourceforge.pmd.lang.vf.ast.ASTAttribute;
+import net.sourceforge.pmd.lang.vf.ast.ASTElExpression;
+import net.sourceforge.pmd.lang.vf.ast.ASTElement;
+import net.sourceforge.pmd.lang.vf.ast.ASTIdentifier;
+import net.sourceforge.pmd.lang.vf.rule.AbstractVfRule;
+
+/**
+ * @author sergey.gorbaty
+ *
+ */
+public class VfCsrfRule extends AbstractVfRule {
+
+    private static final String APEX_PAGE = "apex:page";
+
+    @Override
+    public Object visit(ASTElement node, Object data) {
+        if (APEX_PAGE.equalsIgnoreCase(node.getName())) {
+            List<ASTAttribute> attribs = node.findChildrenOfType(ASTAttribute.class);
+            boolean controller = false;
+            boolean action = false;
+            boolean isEl = false;
+            ASTElExpression valToReport = null;
+
+            for (ASTAttribute attr : attribs) {
+                switch (attr.getName().toLowerCase()) {
+                case "action":
+                    action = true;
+                    break;
+                case "controller":
+                    controller = true;
+                    break;
+                default:
+                    break;
+
+                }
+
+                if (action) {
+                    ASTElExpression value = attr.getFirstDescendantOfType(ASTElExpression.class);
```

Comment:
shouldn't this block be directly within the `case "action":` block? I believe you just want to check the value of the `action` attribute, not all attributes after the `action` one